tool/Makefile

```diff
@@ -12,29 +12,41 @@ OBJS = *.o
 
 CFLAGS += -pthread -I ../include
 
+ifdef DEBUG
+CFLAGS += -g -O0
+endif
+
 LDFLAGS =
 
 LIBS = 
 
 PREFIX = /usr/local
 
+DIR := ${CURDIR}
+LIBDIR := ""
+ifndef DESTDIR
+	LIBDIR=$(DIR)/../lib/
+else
+	LIBDIR=$(DESTDIR)/lib
+endif
+
 tlspool_LIB = ../lib/libtlspool.so
-tlspool_LDFLAGS = -L ../lib -ltlspool
+tlspool_LDFLAGS = -L ../lib -ltlspool -Wl,-rpath,$(LIBDIR)
```

Comment:
maybe we should only use it for 'in source builds' then. this is importatnt since `tools/testcli` would be using the wrong `tlspool.so` when developing.

if the DESTDIR/PREFIX was changed we should update all dependent targets, therefore:

* if it was not set, we are doing 'in source builds' for hacking/testing
* if it was set to DESTDIR=/ it is set for installation and then we should not use RPATH

what would you think about that?